components/external_server/src/model/issue_tracker.py

```diff
@@ -21,6 +21,32 @@ def as_dict(self) -> dict[str, str]:
         return asdict(self)  # pragma: no cover behave
 
 
+@dataclass
+class IssueTrackerCredentials:
+    """Issue tracker credentials needed to create issues."""
+
+    username: str = ""
+    password: str = ""
+    private_token: str = ""
+
+    def basic_auth_credentials(self) -> tuple[str, str] | None:
+        """Return the basic authentication credentials, if any."""
+        return (self.username, self.password) if self.username or self.password else None
+
+    def auth_headers(self) -> dict[str, str]:
+        """Return the authorization headers, if any."""
+        return dict(Authorization=f"Bearer {self.private_token}") if self.private_token else {}
```

Comment:
Currently we always send both and let Jira deal with it. So far that seems to work fine, either because users know they should provide either username/password or token, or because Jira knows what to do. Anyway, we really should improve the UI first to make it impossible for users to provide both types of credentials.